app/api/tax.py

```diff
@@ -66,10 +69,33 @@ def before_create_object(self, data, view_kwargs):
 
 
 class TaxDetail(ResourceDetail):
+    """
+     tax detail by id
+    """
+    def before_get_object(self, view_kwargs):
+        if view_kwargs.get('event_identifier'):
```

Comment:
@shubham-padia Resolved. It was already implemented like that at several other places. Changed all those endpoints affeted by this PR. More changes are to made when all `id` is replaced by `event_id`. Please check now.